src/Toggle Button/main.vala

```diff
@@ -0,0 +1,18 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+
+public void main () {
+
+    string[] button_ids = { "button_no_look", "button_look", "button_camera", "button_flashlight", "button_console" };
+    string[] button_names = { "Don't look", "Look", "Camera", "Flashlight", "Console" };
+
+    for (int i = 0; i < button_ids.length; i++) {
+        string id = button_ids[i];
+        string name = button_names[i];
+        var button = (Gtk.ToggleButton) workbench.builder.get_object (id);
+
+        button.notify["active"].connect (() => {
+            bool active = button.active;
+            message (@"$name $(active ? "On " : "Off ")");
```

Comment:
I'm not a fan of nesting a ternary operator inside the formatted string. I'd go for a similar approach to the Rust demo.

```vala
string status = button.active ? "On" : "Off";
message (@"$name $status");
```